docs/source/starter/lightning_lite.rst

```diff
@@ -0,0 +1,721 @@
+###########################################
+LightningLite - Stepping Stone to Lightning
+###########################################
+
+
+.. image:: https://pl-public-data.s3.amazonaws.com/docs/static/images/lite/lightning_lite.gif
+    :alt: Animation showing how to convert a standard training loop to a Lightning loop
+    :width: 600px
+    :align: center
+
+|
+
+:class:`~pytorch_lightning.lite.LightningLite` enables pure PyTorch users to scale their existing code
+on any kind of device while retaining full control over their own loops and optimization logic.
+
+:class:`~pytorch_lightning.lite.LightningLite` is the right tool for you if you match one of the two following descriptions:
+
+- I want to quickly scale my existing code to multiple devices with minimal code changes.
+- I would like to convert my existing code to the Lightning API, but a full path to Lightning transition might be too complex. I am looking for a stepping stone to ensure reproducibility during the transition.
+
+
+.. note:: LightningLite is a Lightning beta feature and will be considered stable for v1.6.
+
+----------
+
+
+****************
+Learn by example
+****************
+
+
+My existing PyTorch code
+========================
+
+The ``run`` function contains custom training loop used to train ``MyModel`` on ``MyDataset`` for ``num_epochs`` epochs.
+
+.. code-block:: python
+
+    import torch
+    from torch import nn
+    from torch.utils.data import DataLoader, Dataset
+
+
+    class MyModel(nn.Module):
+        ...
+
+
+    class MyDataset(Dataset):
+        ...
+
+
+    def run(args):
+
+        device = "cuda" if torch.cuda.is_available() else "cpu"
+
+        model = MyModel(...).to(device)
+        optimizer = torch.optim.SGD(model.parameters(), ...)
+
+        dataloader = DataLoader(MyDataset(...), ...)
+
+        model.train()
+        for epoch in range(args.num_epochs):
+            for batch in dataloader:
+                batch = batch.to(device)
+                optimizer.zero_grad()
+                loss = model(batch)
+                loss.backward()
+                optimizer.step()
+
+
+    run(args)
+
+
+----------
+
+
+Convert to LightningLite
+========================
+
+Here are 5 required steps to convert to :class:`~pytorch_lightning.lite.LightningLite`.
+
+1. Subclass :class:`~pytorch_lightning.lite.LightningLite` and override its :meth:`~pytorch_lightning.lite.LightningLite.run` method.
+2. Move the body of your existing `run` function.
+3. Remove all calls to `to`, `cuda` etc. as `LightningLite` will take care of it
+4. Apply :meth:`~pytorch_lightning.lite.LightningLite.setup` over each model and optimizers pair, :meth:`~pytorch_lightning.lite.LightningLite.setup_dataloaders` on all your dataloaders and replace ``loss.backward()`` by ``self.backward(loss)``
+5. Instantiate your :class:`~pytorch_lightning.lite.LightningLite` and call its :meth:`~pytorch_lightning.lite.LightningLite.run` method.
```

Comment:
```suggestion
1. Subclass :class:`~pytorch_lightning.lite.LightningLite` and override its :meth:`~pytorch_lightning.lite.LightningLite.run` method.
2. Move the body of your existing `run` function.
3. Remove all `.to`, `.cuda` etc calls since `LightningLite` will take care of it.
4. Apply :meth:`~pytorch_lightning.lite.LightningLite.setup` over each model and optimizers pair, :meth:`~pytorch_lightning.lite.LightningLite.setup_dataloaders` on all your dataloaders and replace ``loss.backward()`` by ``self.backward(loss)``.
5. Instantiate your :class:`~pytorch_lightning.lite.LightningLite` and call its :meth:`~pytorch_lightning.lite.LightningLite.run` method.
```

Reply:
Done !